### PR TITLE
Extend docs for testing plugin code

### LINKED ISSUE #971: Document pytester plugin
Create documentation on how to use the `pytester` plugin. 

It is an invaluable tool for plugin authors, so describing the high level API and linking it from `writing_plugins` would be a nice addition to the documentation.

### PR DESCRIPTION
**Pull request by hackebrot**
Resolve #971 📝 😄

### CODE REVIEW

**Comment by The-Compiler** [comment]
Great work Raphael! 🎉 

Seeing that this only changes documentation, and doesn't rely on anything new in the `features` branch, shouldn't it go to `master`?

File: doc/en/writing_plugins.rst [hunk 1]
```diff
@@ -286,34 +286,73 @@ the ``--trace-config`` option.
...
+                return request.param
         """)
-        result = testdir.runpytest("--verbose")
-        result.stdout.fnmatch_lines("""
-            test_example*
+
+        # create a temporary pytest test file
+        testdir.makepyfile("""
+            def test_hello_default(hello):
+                assert hello() == "Hello World!"
```
Comment (nicoddemus) [inline doc/en/writing_plugins.rst]: Hmm I don't see where the `hello` fixture is defined.

File: doc/en/writing_plugins.rst [hunk 2]
```diff
@@ -286,34 +286,73 @@ the ``--trace-config`` option.
...
-in-process.  You can pass the command line option ``--runpytest=subprocess``
-to have it happen in a subprocess.
+        # run all tests with pytest
+        result = testdir.runpytest()
+
+        # check that all 4 tests passed
+        result.assert_outcomes(passed=4)
+
+
+For more information about the result object, that ``runpytest()`` returns, and
```
Comment (nicoddemus) [inline doc/en/writing_plugins.rst]: typo: "result object, that" -> "result object that"

File: doc/en/writing_plugins.rst (hunk 1 above)
Comment (The-Compiler) [inline doc/en/writing_plugins.rst]: In the (not displayed) plugin which is tested was my guess.

Comment (hackebrot) [inline doc/en/writing_plugins.rst]: It's provided by our plugin.

**Comment by hackebrot** [comment]
Thank you @The-Compiler for pointing out that this should target master. 🙇 

Closing this in favor of #2593.

**Comment by The-Compiler** [comment]
btw, you can [change the branch](https://help.github.com/articles/changing-the-base-branch-of-a-pull-request/) of an existing pull request too :wink:

**Comment by hackebrot** [comment]
@The-Compiler Not in this case. There were merge conflicts and I generally prefer to be in control of git stuff, no idea what exactly the GUI is doing, but it's certainly not a clean rebase 😛

File: doc/en/writing_plugins.rst (hunk 1 above)
Comment (nicoddemus) [inline doc/en/writing_plugins.rst]: Hmm I see. Perhaps it would be clear to provide the full example instead?

Comment (hackebrot) [inline doc/en/writing_plugins.rst]: LOL 😂 Just added the plugin code. Please review #2593 

I can revert the last commit, np. Let me now! ☺️

**Comment by The-Compiler** [comment]
IIRC it doesn't do any rebasing - you'd need to rebase and force-push your branch after changing the PR target branch.

File: doc/en/writing_plugins.rst (hunk 1 above)
Comment (nicoddemus) [inline doc/en/writing_plugins.rst]: Heh no worries, it was not required but it did improve things IMHO. 😉

**Pull request closed without merging**